crates/bevy_ecs/src/lib.rs

```diff
@@ -1262,6 +1266,36 @@ mod tests {
         assert_eq!(world.resource::<A>().0, 1);
     }
 
+    #[test]
+    fn non_send_resource_scope() {
+        let mut world = World::default();
+        world.insert_non_send_resource(NonSend::default());
+        world.resource_scope(|world: &mut World, mut value: Mut<NonSend>| {
+            value.0 += 1;
+            assert!(!world.contains_resource::<NonSend>());
+        });
+        assert_eq!(world.non_send_resource::<NonSend>().0, 1);
+    }
+
+    #[test]
+    #[should_panic(expected = "attempted to access NonSend resource bevy_ecs::tests::NonSend off of the main thread")]
+    fn non_send_resource_scope_from_different_thread() {
+        let mut world = World::default();
+        world.insert_non_send_resource(NonSend::default());
+
+        let thread = std::thread::spawn(move || {
+            // Accessing the non-send resource on a different thread
+            // Should result in a panic
+            world.resource_scope(|_: &mut World, mut value: Mut<NonSend>| {
+                value.0 += 1;
+            });
+        });
+
+        if let Err(err) = thread.join() {
+            panic!("{}", err.downcast::<String>().unwrap());
+        }
```

Comment:
I tried to do it that way, but then the panic message doesn't match the expected one ("called `Result::unwrap()` on an `Err` value: Any { .. }" instead of "attempted to access NonSend resource bevy_ecs::tests::NonSend off of the main thread").

Reply:
It's sad that that doesn't contain the actual error message. This approach looks fine then. Just made a small suggestion above.